sound/drivers/xen-sndfront.c

```diff
@@ -0,0 +1,2127 @@
+/*
+ *  Xen para-virtual sound device
+ *
+ *  Based on sound/drivers/dummy.c
+ *  Based on drivers/net/xen-netfront.c
+ *  Based on drivers/block/xen-blkfront.c
+ *
+ *   This program is free software; you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation; either version 2 of the License, or
+ *   (at your option) any later version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ * Copyright (C) 2016 EPAM Systems Inc.
+ */
+
+#include <linux/kernel.h>
+#include <linux/errno.h>
+#include <linux/module.h>
+#include <linux/slab.h>
+#include <linux/platform_device.h>
+#include <linux/irq.h>
+#include <linux/vmalloc.h>
+
+#include <sound/core.h>
+#include <sound/pcm.h>
+
+#include <asm/xen/hypervisor.h>
+#include <xen/xen.h>
+#include <xen/platform_pci.h>
+#include <xen/xenbus.h>
+#include <xen/events.h>
+#include <xen/grant_table.h>
+#include <xen/interface/io/ring.h>
+#include <xen/interface/io/sndif_linux.h>
+
+#define GRANT_INVALID_REF	0
+/* timeout in ms to wait for backend to respond */
+#define VSND_WAIT_BACK_MS	5000
+
+enum xdrv_evtchnl_state {
+	EVTCHNL_STATE_DISCONNECTED,
+	EVTCHNL_STATE_CONNECTED,
+	EVTCHNL_STATE_SUSPENDED,
+};
+
+struct xdrv_evtchnl_info {
+	struct xdrv_info *drv_info;
+	struct xen_sndif_front_ring ring;
+	int ring_ref;
+	unsigned int port;
+	unsigned int irq;
```

Comment:
Why unsigned int? Type int is used in both cases "int bind_evtchn_to_irqhandler()" and "notify_remote_via_irq(int)"